main/src/main/scala/sbt/Plugins.scala

```diff
@@ -341,17 +341,15 @@ ${listConflicts(conflicting)}""")
     }
 
   private[sbt] def hasAutoImportGetter(ap: AutoPlugin, loader: ClassLoader): Boolean = {
-    import reflect.runtime.{universe => ru}
     import util.control.Exception.catching
-    val m = ru.runtimeMirror(loader)
-    val im = m.reflect(ap)
-    val hasGetterOpt = catching(classOf[ScalaReflectionException]) opt {
-      im.symbol.asType.toType.declaration(ru.newTermName("autoImport")) match {
-        case ru.NoSymbol => false
-        case sym => sym.asTerm.isGetter || sym.asTerm.isModule
-      }
+    val pluginClazz = ap.getClass
+    val autoImportExists = catching(classOf[ClassNotFoundException]).opt {
+      val autoImportClassName = s"${pluginClazz.getName}autoImport$$"
+      val fields = pluginClazz.getDeclaredFields
```

Comment:
This should walk up the superclass chain in case the `AutoPlugin` inherits the `autoImport` member.

We no longer can check whether or not `autoImport` is a stable term (`val` or `object`), but I think it is fine to document that plugin authors must not declare `var autoImport`.